File: Unit4E2.go
// Author: William
// Version: 1.0.0
// Date: 2025-11-25
// Fileoverview: This program asks for a positive integer and counts down to 0.

package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {

	var number int
	reader := bufio.NewReader(os.Stdin)

	// input loop (do-while equivalent)
	for {
		fmt.Print("Enter a positive integer: ")
		input, _ := reader.ReadString('\n')
		input = strings.TrimSpace(input)
		number, _ = strconv.Atoi(input)

		if number > 0 {
			break
		}

		fmt.Println("Invalid input. Please enter a positive integer:")
	}

	// countdown
	fmt.Printf("Counting down from %d:\n", number)

	for number >= 0 {
		fmt.Println(number)
		number--
	}

	fmt.Println("\nDone.")
}

File: Unit4E2.ts
/**
 * @author William
 * @version 1.0.0
 * @date 2025-11-27
 * @fileoverview This program asks for a positive integer and counts down to 0.
 */

// variables
let number: number = 0;

// input loop
do {
  const userInput = prompt("Enter a positive integer: ") || "0";
  number = parseInt(userInput);

  if (number <= 0) {
    console.log("Invalid input. Please enter a positive integer:");
  }
} while (number <= 0);

// countdown
console.log("Counting down from " + number + ":");

while (number >= 0) {
  console.log(number);
  number = number - 1;
}

console.log("\nDone.");
